R/stan.R

```diff
@@ -153,3 +153,50 @@ fit_TIRT_stan <- function(data, init = 0, ...) {
   structure(nlist(fit, data), class = "TIRTfit")
 }
 
+#' @export
+get_TIRT_params_stan <- function(fitted_model) {
+  # extract fit parameters from the stanmodel object
+  components = rstan::extract(fitted_model)
+  # person parameters
+  fit_eta = apply(components$eta,c(2,3),mean)
+  colnames(fit_eta) = sprintf("trait%s",seq(1:ncol(fit_eta)))
+  fit_lambda = apply(components$lambda,2,mean)
+  fit_Phi = apply(components$Cor_trait,c(2,3),mean)
+  fit_gamma = apply(components$gamma,2,mean)
+  fit_psi = apply(components$psi,2,mean)
+  fit_r = apply(components$r,2,mean)
+  out = list(lambda = fit_lambda,
+             phi = fit_Phi,
+             gamma = fit_gamma,
+             psi = fit_psi,
+             eta = fit_eta,
+             r = fit_r)
+  return(out)
+}
+
+get_prediction_model <- function(fitted_model) {
```

Comment:
I am a little confused about the purpose of the function. Specifically, there seems to be a local path involved. Is this function really necessary or can we just copy the code over to `fit_TIRTpredictor_stan`?